src/common/ipc/cluster/index.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * This channel is broadcast on whenever the cluster fails to list namespaces
+ * during a refresh and no `accessibleNamespaces` have been set.
+ */
+export const ClusterListNamespaceForbiddenChannel = "cluster:list-namespace-forbidden";
```

Comment:
This file contains parts of `type-enforced-ipc.ts`, right?
Is there any special reason why splitting those? Can we wrap it in single "action" an export to just call outside?
Also I think `onCorrect` is too way abstract in general.

Reply:
Do you mean instead of the raw broadcast and receiving pair being exported, just other items? 

What do you mean exactly bu `onCorrect` (and I assume `onceCorrect` too) are way too abstract? Yeah the types assume that you ignore the first arg but other than that they do the type checking that such an call should have done to it.

Reply:
I think a different PR is in order for that work to be done, since I don't really see it as part of this PR. But yes I think I can rearrange the code to be easier.